### ISSUE TITLE
Cannot wipe a file system on a RAID device

### ISSUE THREAD

**Issue by nettings** [openmediavault/openmediavault#1222]
Fresh installation of omv6 on Armbian bullseye (odroidhc4).
System running from SD card, with two harddrives which already contain a previous RAID1 device.

Objective is to wipe the RAID and create a fresh file system on an encrypted LUKS device on said RAID1.

I found that /dev/md0 doesn't show up in the devices dropdown of the Encryption "create new" section, presumably because omv6 assumes it is in use.

Tried the following:
* delete RAID1 device
* quick-wipe /dev/sda
* quick-wipe /dev/sdb
* recreate RAID1 device

/dev/md0 still doesn't show up. Turns out that quick-wipe only nukes the partition table - the file system signature is still valid, and I could even mount /dev/md0 afterwards, as its previous ext4 fs had survived.

The only way around this was to physically destroy the file system by doing `dd if=/dev/zero of=/dev/md0 bs=512 count=1000000".
After doing this, the RAID device shows up in the Encryption device selector.

I guess the bug is that we need a GUI wipe feature on RAID devices as well as on physical harddisks. Doing a secure wipe would likely have worked, too, but you don't really want to do that on a 10TB drive if you don't have to.

**Comment by ryecoaaron** [comment]
The only true way to get all filesystem signatures is to wipe the full thing.  Your dd command might get most but not all.  If you create the array from the command line, it would prompt you to overwrite the filesystem signature but that can't be passed to the web interface.  As far as I know, there is no way to pass a parameter that tells it is ok to wipe the signature on new array creation.

**Comment by votdev** [comment]
> Turns out that quick-wipe only nukes the partition table 

That's hard to imagine because the quick wipe does zap the disk and erases the first and last 4MiB of the disk. See https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/share/openmediavault/engined/rpc/diskmgmt.inc#L243 for the code. Any suggestions how to improve it?

**Comment by votdev** [comment]
By the way, OMV6 removed the feature to erase an file system. The new approach here is to be non destructive. If you wantto delete the file system on a MD RAID, you need to do it on your own via CLI. The disk wipe feature does not work for MD devices.

**Comment by nettings** [comment]
@votdev: Funny... like I said, I was able to mount the previously existing ext4 on /dev/md0 on the command line after removing the RAID, quick-wiping both disks that belonged to that array, and recreating it. And there was even data on it, and no obvious errors. Maybe the 4MiB will kill the partition table and the RAID header, but leave the fs structure intact?
@ryecoaaron, I must confess I don't know how RAID information is stored - if it has something like the redundant superblocks of some filesystems, then I would understand how the FS signature itself survived. I was assuming I killed the RAID, and the newly created RAID "found" the remnant ext4...
In any case, if you say that those heavily destructive operations are out of scope for the OMV GUI, that makes sense to me.

But then it would be nice to have more documentation on which operations are safe to perform as root via the CLI - I was reluctant to try it because numerous forum and documentation articles warn against it. Is it generally safe to assume that all filesystem-related information is gathered on the fly by OMV when needed, or do some parts of it end up in omv's database and could go out of sync with reality if I hack away on the CLI like the old linux geezer that I am?
